### PR TITLE
Add Spark 3.2.4 shim

### LINKED ISSUE #7094: [FEA] Add a shim layer for Spark 3.2.4
The branch-3.2 is now Spark 3.2.4, a shim layer is needed to support it.

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds Spark 3.2.4 Shim

fixes #7094

### CODE REVIEW

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
Converting this to draft until we merge the private repo changes

File: sql-plugin/src/main/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimServiceProvider.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline sql-plugin/src/main/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimServiceProvider.scala:2]: remove 2022

File: sql-plugin/src/main/spark324/scala/com/nvidia/spark/rapids/spark324/RapidsShuffleManager.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline sql-plugin/src/main/spark324/scala/com/nvidia/spark/rapids/spark324/RapidsShuffleManager.scala:2]: remove 2022

File: sql-plugin/src/main/spark324/scala/org/apache/spark/sql/rapids/shims/spark324/RapidsShuffleInternalManager.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline sql-plugin/src/main/spark324/scala/org/apache/spark/sql/rapids/shims/spark324/RapidsShuffleInternalManager.scala:2]: remove 2022

File: sql-plugin/src/test/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimsSuite.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```
Comment (tgravescs) [inline sql-plugin/src/test/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimsSuite.scala:2]: remove 2022

**Review by tgravescs — COMMENTED** [review]
you also need to update table in https://github.com/NVIDIA/spark-rapids/blob/branch-23.06/docs/additional-functionality/rapids-shuffle.md

**Comment by razajafri** [comment]
This PR is ready to go in. I will kick off the CI

**Comment by razajafri** [comment]
build

**Review by tgravescs — APPROVED** [review]

File: sql-plugin/src/test/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimsSuite.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "324"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims.spark324;
```
Comment (gerashegalov) [inline sql-plugin/src/test/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimsSuite.scala:20]: nit: we keep replicating extra semicolon, for this or a future PR could get rid of all of them.
```
$ git grep -c '^package .*;' '*.scala' | wc -l
14
```

```suggestion
package com.nvidia.spark.rapids.shims.spark324
```

File: pom.xml [hunk 6]
```diff
@@ -280,6 +280,31 @@
                 <module>aggregator</module>
             </modules>
         </profile>
+        <profile>
+            <id>release324</id>
+            <activation>
+                <property>
+                    <name>buildver</name>
+                    <value>324</value>
```
Comment (gerashegalov) [inline pom.xml:288]: this needs to be included into `all.buildvers`

**Review by gerashegalov — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 6 above)
Comment (NVnavkumar) [reply pom.xml:288]: I think `all.buildvers` is defined to include `noSnapshot.buildvers`, correct?

```
        <all.buildvers>
            ${noSnapshot.buildvers},
            ${snapshot.buildvers},
            ${databricks.buildvers},
        </all.buildvers>
```

Comment (razajafri) [reply pom.xml:288]: Yes, it does https://github.com/NVIDIA/spark-rapids/blob/ea89dd58cb101834703da4e3b2b807cd4f0fbb6f/pom.xml#L706

File: sql-plugin/src/test/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimsSuite.scala (hunk 5 above)
Comment (razajafri) [reply sql-plugin/src/test/spark324/scala/com/nvidia/spark/rapids/shims/spark324/SparkShimsSuite.scala]: I can create a cleanup PR for the other files

File: pom.xml (hunk 6 above)
Comment (gerashegalov) [reply pom.xml:288]: it does but I overlooked the line https://github.com/NVIDIA/spark-rapids/pull/8239/commits/d4e7063abb44d9381e20f85901292b47411f118d#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R699

**Review by gerashegalov — APPROVED** [review]
LGTM

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Review by NvTimLiu — APPROVED** [review]
LGTM

**Review by jlowe — APPROVED** [review]

**Pull request merged**